src/python/ensembl/io/genomio/fasta/chunk.py

```diff
@@ -78,6 +127,198 @@ def split_seq_by_chunk_size(ends: list[int], chunk_size: int, tolerated_size: Op
+        chunk_size: Size of the chunks to split into.
+        chunk_size_tolerated: If more flexibility allowed, use this as the maximum size of a chunk.
+        output_fasta: Output FASTA TextIOWrapper stream to store the chunks into,
+                if none or False, `open_individual` is used (see below).
+        individual_file_prefix: A file path prefix including dirs and filenames part to use as a
+                firts part of the chunk file name.
+        n_sequece_len: Length of the stretch of `N`s to split at; not slitting on `N`s if 0.
+        chunk_sfx: A string to put between the original sequence name and the chunk suffix.
+        append_offset_to_chunk_name: A flag to append 0-based offset (`_off_{offset}`) to the chunk name.
+        open_individual: A callable taking filename as an input to generatethe output file for
+                individual contig if out_put FASTA is `false` of `None`, folders should be preexisting.
+    """
+
+    chunk_size_tolerated = max(chunk_size, chunk_size_tolerated)
+    # output agp_lines list
+    agp_lines = []
+
+    # make sure not used for n_seq <= 0
+    n_split_regex = None
+    if n_sequece_len > 0:
+        pattern = f"(N{{{n_sequece_len},}})"
+        n_split_regex = re.compile(pattern)
+
+    # process stream
+    fasta_parser = SeqIO.parse(input_fasta, "fasta")
+    for rec_count, rec in enumerate(fasta_parser, start=1):
+        rec_name = str(rec.name)
+
+        ends = split_seq_by_n(str(rec.seq), n_split_regex)
+        ends = split_seq_by_chunk_size(ends, chunk_size, chunk_size_tolerated)
+
+        offset = 0
+        for chunk, chunk_end in enumerate(ends, start=1):
+            chunk_name = f"{rec_name}_{chunk_sfx}_{chunk:03d}"
+            chunk_file_name = ""
+            if individual_file_prefix:
+                chunk_file_name = f"{individual_file_prefix}.{rec_count:03d}.{chunk:03d}.fa"
+            if append_offset_to_chunk_name:
+                chunk_name += f"_off_{offset}"
+
+            rec_from = offset + 1
+            rec_to = chunk_end
+            chunk_len = chunk_end - offset
+
+            # form agp lines
+            agp_line = f"{rec_name}\t{rec_from}\t{rec_to}\t{chunk}\tW\t{chunk_name}\t1\t{chunk_len}\t+"
+            agp_lines.append(agp_line)
+
+            # use agp lines as fasta description
+            agp_line = agp_line.replace("\t", " ")
+            logging.info(f"Dumping {chunk_name} AGP {agp_line}")
+
+            # get slice and put it out
+            tmp_record = SeqRecord(
+                Seq(rec.seq[offset:chunk_end]),
+                id=chunk_name,
+                description=f"AGP {agp_line}",
+                name="",
+            )
+
+            # if user specified chunks -- store each chunk in an individual output file
+            with output_fasta and nullcontext(output_fasta) or open_individual(chunk_file_name) as out_file:
+                out_file.write(tmp_record.format("fasta"))  # type: ignore[union-attr]
```

Comment:
```suggestion
                SeqIO.write([tmp_record], out_file, "fasta")
```
I find this option more readable (unfortunately the function expects an iterator, so the list is unavoidable).